### ISSUE TITLE
Query with MIN on collected variable crashes Db

### ISSUE THREAD

**Issue by JosipVuckovic** [arangodb/arangodb#14807]
## My Environment

* __ArangoDB Version__:        3.8.1
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   ArangoDB Starter
* __Configuration__:               
* __Infrastructure__:               own
* __Operating System__:      Windows 10 
* __Total RAM in your machine__:        
* __Disks in use__:        SSD
* __Used Package__:              other

## Component, Query & Data

__Affected feature__:
ArangoDb instance

__AQL query (if applicable)__:

```
for u in union([],[])
                collect test = u.v into g
                return Distinct {
                    Test: test,
                    Prop1: MIN(g[*].u.someProp),
                    Prop2: MAX(g[*].u.someProp)
                }
```

__AQL explain and/or profile (if applicable)__:

```
Execution plan:
 Id   NodeType            Est.   Comment
  1   SingletonNode          1   * ROOT
  2   CalculationNode        1     - LET #7 = [ ]   /* json expression */   /* const assignment */
  3   EnumerateListNode      0     - FOR u IN #7   /* list iteration */
  4   CalculationNode        0       - LET #9 = u.`v`   /* attribute expression */
  9   SortNode               0       - SORT #9 ASC   /* sorting strategy: standard */
  5   CollectNode            0       - COLLECT test = #9 INTO g KEEP u   /* sorted */
  6   CalculationNode        0       - LET #11 = { "Test" : test, "Prop1" : MIN(g[*].`u`.`someProp`), "Prop2" : MAX(g[*].`u`.`someProp`) }   /* simple expression */
  7   CollectNode            0       - COLLECT #13 = #11   /* distinct */
  8   ReturnNode             0       - RETURN #13
Indexes used:
 none
Functions used:
 Name   Deterministic   Cacheable   Uses V8
 MIN    true            true        false
 MAX    true            true        false
Optimization rules applied:
 Id   RuleName
  1   remove-collect-variables
Optimization rules with highest execution times:
 RuleName                                    Duration [s]
 remove-collect-variables                         0.00002
 specialize-collect                               0.00002
 remove-redundant-calculations                    0.00001
 use-indexes                                      0.00001
 optimize-count                                   0.00000
41 rule(s) executed, 1 plan(s) created
```

__Dataset__:
N/A

__Size of your Dataset on disk__:
minimal

__Replication Factor & Number of Shards (Cluster only)__:
N/A

## Steps to reproduce

1. Just run the AQL query
2. 
3. 

__Problem__:
Whole ArangoDb instance crashes, i.e. I need to manually restart the ArangoDb service.

__Expected result__:
Load data

**Comment by goedderz** [comment]
Hi @JosipVuckovic,

I tried to reproduce it, but cannot: I get an empty result set as expected.

Is there something in the arangod log files that you can share?

Best,

Tobias

**Comment by JosipVuckovic** [comment]
The log file was empty, on the machine I am running 3.7.2 I also got the empty list as expected, but on this with 3.8.1. it keeps crashing.

**Comment by Simran-B** [comment]
I can confirm that running the query in 3.8.1 crashes on Windows 10, but the Docker container does not.

`{crash} Unhandled exception: c0000005 at address 00007ff6192b8ab0 in thread 407c`

Here's what WinDbg reported:

```
arangod!std::_Fnv1a_append_bytes [inlined in arangod!std::_Fnv1a_append_value<unsigned int>]:
00007ff6`192b8ab0 440fb602        movzx   r8d,byte ptr [rdx] ds:00000000`00000000=??
```

```
[0x0]   arangod!std::_Fnv1a_append_bytes   
[0x1]   arangod!std::_Fnv1a_append_value<unsigned int>   
[0x2]   arangod!std::hash<unsigned int>::_Do_hash + 0x8   
[0x3]   arangod!std::_Conditionally_enabled_hash<unsigned int,1>::operator() + 0x8   
[0x4]   arangod!emilib::HashSet<unsigned int,std::hash<unsigned int>,emilib::HashSetEqualTo<unsigned int> >::find_filled_bucket + 0x12   
[0x5]   arangod!emilib::HashSet<unsigned int,std::hash<unsigned int>,emilib::HashSetEqualTo<unsigned int> >::find + 0x12   
[0x6]   arangod!arangodb::aql::removeCollectVariablesRule + 0x3a3   
[0x7]   arangod!arangodb::aql::Optimizer::createPlans + 0x29a   
[0x8]   arangod!arangodb::aql::Query::preparePlan + 0x32b   
[0x9]   arangod!arangodb::aql::Query::prepareQuery + 0x4e   
[0xa]   arangod!arangodb::aql::Query::execute + 0x42d   
[0xb]   arangod!arangodb::RestCursorHandler::processQuery + 0x56   
[0xc]   arangod!arangodb::RestCursorHandler::registerQueryOrCursor + 0x762   
[0xd]   arangod!arangodb::RestCursorHandler::createQueryCursor + 0xbf   
[0xe]   arangod!arangodb::rest::RestHandler::executeEngine + 0xac   
[0xf]   arangod!arangodb::rest::RestHandler::runHandlerStateMachine + 0x10c   
[0x10]   arangod!arangodb::rest::RestHandler::runHandler + 0x85   
[0x11]   arangod!arangodb::rest::CommTask::handleRequestAsync::__l8::<lambda_c796163337351d47c76198ea236ce46f>::operator() + 0x51   
[0x12]   arangod!arangodb::Scheduler::WorkItem<<lambda_c796163337351d47c76198ea236ce46f> >::invoke + 0x5a   
[0x13]   arangod!arangodb::SupervisedScheduler::runWorker + 0x247   
[0x14]   arangod!arangodb::Thread::runMe + 0x34   
[0x15]   arangod!arangodb::Thread::startThread + 0xc0   
[0x16]   arangod!ThreadStarter + 0x1b   
[0x17]   KERNEL32!BaseThreadInitThunk + 0x14   
[0x18]   ntdll!RtlUserThreadStart + 0x21   
```

**Comment by JosipVuckovic** [comment]
@Simran-B Thank you for the information, we can then use docker as a gap solution.

**Comment by goedderz** [comment]
@mpoeter and I found the bug, a bugfix is prepared (see https://github.com/arangodb/arangodb/pull/14824 / 
https://github.com/arangodb/arangodb/pull/14825).

**Comment by goedderz** [comment]
@JosipVuckovic The fix was merged and will be released with 3.8.2.

**Comment by JosipVuckovic** [comment]
@goedderz 
Thank you for the info and quick fix.
